modin/pandas/utils.py

```diff
@@ -252,6 +254,88 @@ def check_both_not_none(option1, option2):
     return not (option1 is None or option2 is None)
 
 
+def reindex_and_broadcast_item(
+    obj,
+    row_lookup,
+    col_lookup,
+    item,
+    need_columns_reindex=True,
+):
+    """
+    Use NumPy to broadcast or reshape item with reindexing.
+
+    Parameters
+    ----------
+    obj : DataFrame or Series
+        The object contains the necessary information about the axes.
+    row_lookup : slice or scalar
+        The global row index to locate inside of `item`.
+    col_lookup : range, array, list, slice or scalar
+        The global col index to locate inside of `item`.
+    item : DataFrame, Series, or query_compiler
+        Value that should be broadcast to a new shape of `to_shape`.
+    need_columns_reindex : bool, default: True
+        In the case of assigning columns to a dataframe (broadcasting is
+        part of the flow), reindexing is not needed.
+
+    Returns
+    -------
+    numpy.ndarray
+        `item` after it was broadcasted to `to_shape`.
+
+    Raises
+    ------
+    ValueError
+        If `row_lookup` or `col_lookup` contain values missing in
+        DataFrame/Series index or columns correspondingly.
+        If `item` cannot be broadcast from its own shape to `to_shape`.
+
+    Notes
+    -----
+    NumPy is memory efficient, there shouldn't be performance issue.
+    """
+    # It is valid to pass a DataFrame or Series to __setitem__ that is larger than
+    # the target the user is trying to overwrite. This
```

Comment:
```suggestion
    # the target the user is trying to overwrite.
```
Should we remove or extend this?